### ISSUE TITLE
markdownlint.yaml is not interpreted the same as markdownlint-cli2

### ISSUE THREAD

**Issue by jankatins** [rvben/rumdl#389]
### rumdl version

0.1.15

### Installation method

Other

### Bug description

I run `uvx rumdl ` on my project which is currently linted by https://github.com/DavidAnson/markdownlint-cli2

According to https://rumdl.dev/markdownlint-comparison/#4-configuration-philosophy, the `.markdownlint.yaml` is used as a config file and this is also visible in the rumdl config output (showing "from project config"). But the config is differently interpreted than markdownlint.-li2.

### Steps to reproduce

 I've boiled it down to the following config:

```shell
λ  cat .markdownlint.yaml
default: true

MD001: true # <- this is the poblematic line!

MD013:
  line_length: 120

λ  npx markdownlint-cli2 README.md
markdownlint-cli2 v0.20.0 (markdownlint v0.40.0)
Finding: README.md
Linting: 1 file(s)
Summary: 19 error(s)
README.md:38:121 error MD013/line-length Line length [Expected: 120; Actual: 223]
... 18 more problems ... 

λ  uvx rumdl check --verbose README.md
Enabled rules:
  - MD001 (Heading levels should only increment by one level at a time)

Exclude patterns: []
Processing file: README.md
Linting took: 2.226445ms
Total processing time for .../README.md: 2.584919ms

Success: No issues found in 1 file (2ms)
```

If I remove the `MD001: true` from the config, it now works:

```shell
λ  uvx rumdl check --verbose README.md
Enabled rules:
  - MD001 (Heading levels should only increment by one level at a time)
  - ...
  - MD013 (Line length should not be excessive)
  - ...
  - MD074 (MkDocs nav entries should point to existing files)

Exclude patterns: []
Processing file: README.md
Linting took: 13.581191ms
Total processing time for .../README.md: 13.964819ms
README.md:38:121: [MD013] Line length 223 exceeds 120 characters
... a lot more issues ....
Loaded workspace index from cache with 1 files
Workspace index: 0 updated, 1 unchanged, 0 pruned (1 total) in 19.887µs
Saved workspace index cache with 1 files

Issues: Found 35 issues in 1 file (19ms)
Run `rumdl fmt` to automatically fix 27 of the 35 issues
```

### Expected behavior

Config is interpreted the same

### Actual behavior

`MDxxx: true` lines seems to trigger that interpreted all proper config sections as not existing.

### Configuration

Only the .markdownlint.yaml file:

```yaml
default: true

MD001: true # <- this is the poblematic line!

MD013:
  line_length: 120
```

### Operating system

Linux

### Additional context

The full `.markdownlint.yaml` file is based on https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.yaml which contains this `MDxxx: true` lines.

**Comment by rvben** [comment]
Fixed in v0.1.16.

The issue was twofold:
1. The `default: true` key in `.markdownlint.yaml` was not handled, causing a parse error that silently affected subsequent rule configs.
2. `MD001: true` (boolean value for a rule) was not recognized as "enable this rule with defaults."

Both cases are now handled correctly — boolean values enable/disable rules, and the `default` key controls whether rules are enabled by default. This matches markdownlint-cli2 behavior.

Thanks for the detailed bug report, @jankatins!

**Comment by jankatins** [comment]
Wow, big thank you for making this compatibility fix!
